VRDR.Messaging/DeathRecordSubmission.cs

```diff
@@ -42,9 +42,12 @@ public DeathRecord DeathRecord
             {
                 deathRecord = value;
                 MessageBundle.Entry.RemoveAll( entry => entry.Resource.ResourceType == ResourceType.Bundle );
-                MessageBundle.AddResourceEntry(deathRecord.GetBundle(), "urn:uuid:" + deathRecord.GetBundle().Id);
                 Header.Focus.Clear();
-                Header.Focus.Add(new ResourceReference(deathRecord.GetBundle().Id));
+                if (deathRecord != null)
+                {
+                    MessageBundle.AddResourceEntry(deathRecord.GetBundle(), "urn:uuid:" + deathRecord.GetBundle().Id);
+                    Header.Focus.Add(new ResourceReference(deathRecord.GetBundle().Id));
```

Comment:
I'll confess I don't fully understand internal reference rules... should urn:uuid: be used here since focus is a reference? Does `ResourceReference` take care of that?

Reply:
Good catch!